### ISSUE TITLE
Replace NSFileManager.enumeratorAtPath with enumeratorAtURL

### ISSUE THREAD

**Issue by dreampiggy** [SDWebImage/SDWebImage#3689]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 11 / Xcode 10
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Using enumeratorAtPath has to keep the `full path string` in memory. Which may cause OOM like #3504 

But however, when using `enumeratorAtURL`, it use relative URL and only keep the filename differences in memory. Which is more performant.

We should change all the call to `enumeratorAtPath` to `enumeratorAtURL`

Apple already point this in documentation: https://developer.apple.com/documentation/foundation/nsfilemanager/1409571-enumeratoraturl

**Comment by ChengzhiHuang** [comment]
I will contribute a PR to this issue.
